### ISSUE TITLE
Fitting message text  in a dialog-window changed

### ISSUE THREAD

**Issue by cage2** [McParen/croatoan#35]
Hi @McParen !

I think something changed in the way a `message-text` is managed in a dialog window.

To show what i mean is sufficient to change these lines (example `t19f`):

https://github.com/McParen/croatoan/blob/master/test/clos.lisp#L2592

``` lisp 
(let* ((items (list "Yes" "No" "OK" 'cancel))
```
to: 

``` lisp 
(let* ((items (list "Yes" "No"))
```

and this one:
https://github.com/McParen/croatoan/blob/master/test/clos.lisp#L2599

``` lisp 
:layout (list 1 4)
```

What we get here is that the message text is fitted (but with words truncated) at a with equal to the cumulative items length instead of fill the windows width.

I do not know if this is actually is a bug or intentional but i am quite sure this behaviour was added in one of the latest commits.

I investigated this a bit and i found that if in this line:

https://github.com/McParen/croatoan/blob/master/src/classes.lisp#L665

we substitute:

``` lisp 
:width (* (cadr layout) item-length)))
```
with:

``` lisp 
:width width
```
The message text expands to fill the window but is is still somehow truncated (something with pad i guess).

If this is indeed a bug i can submit a PR with the changes above, but i am reluctant because i do not think this is enough.
If this is a bug and you are OK to help me with suggestions i can try to get a proper fix.

Bye!
C.

**Comment by McParen** [comment]
Hallo cage,

thanks for the finding. This is clearly a mistake. The intention of the commit was to make the dialog properly render with a without a window border with the curent item prefix taken into account. The default width of the message should depend on the width of the dialog window, not on the number or layout of the buttons, that is true.

Instead of just

    :width width

it should account for a border:

    :width (- width (+ 2 (* 2 padding)))

Do you agree? If you do, I'll push that. If you have a better idea, please tell.

I am in general not all too happy with the pad object containing the message. I did this a long time ago and I think that it can be greatly simplified. The dialog window should in the future be bassed on form-window instead of menu-window, so we can add input fields, checkboxes and buttons. It is only based on menu-window because it was added before I started adding form elements.

**Comment by cage2** [comment]
On Thu, Jul 30, 2020 at 08:03:46AM -0700, McParen wrote:
> Hallo cage,

Hi! :)

I have applied the changes you proposed and it works like a charm! :)

Now i can see what was missed in  my code, i did not took into account
the padding (including border i guess).

> thanks for the finding.

Thanks to  you for your  time spent  in this code,  i was not  able to
write my software without this library (and some others too :-))

I am happy to contribute to it if i can.

> I am in general not all too happy with the pad object containing the
> message. I  did this  a long  time ago and  I think  that it  can be
> greatly simplified. The dialog window should in the future be bassed
> on form-window instead  of menu-window, so we can  add input fields,
> checkboxes and buttons.  It is only based on  menu-window because it
> was added before I started adding form elements.

I think i can  see your point, now that you wrote this  i can see that
the form  seems the "natural"  choice for parent of  window-dialog (as
the last is, as you said, much more powerful and flexible).

Anyway even in the current  hierarchy dialog-window is very useful!
:)

Bye McParen!
C.

**Comment by McParen** [comment]
Fixed in 576460501e7439f4e487f3b6100e9ddf921cde5c
